### PR TITLE
Add guidance for password input component

### PR DESCRIPTION
**Pull request by querkmachine**
WIP of some guidance for the password input component, transposing content from the "ask users for passwords" pattern guidance where it seems relevant.

Related to: 
- #3352
- #3353

Still needs examples.

### CODE REVIEW

**Comment by edwardhorsford** [comment]
If there's isn't a separate pattern for it, it might be worth having a note that mentions secret question inputs - they probably *shouldn't* be treated as password inputs, or at least some of the guidance here isn't correct for them.

It would also be good to have some guidance on 'show password' type functionality.

File: src/components/password-input/index.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+Note that even if `autocomplete` is omitted or set to `off`, many browsers will ignore this and continue to autofill password inputs.
+
+### Allow users to paste their password
+
+Do not disable paste on password fields. People may have very good reasons why they want to paste their password, for example if they’re using a password manager.
+
+### Avoid restricting the user's input
+
+[TODO — maybe just keep this on pattern guidance page?]
```
Comment (edwardhorsford) [inline src/components/password-input/index.md:60]: Might want to explicitly mention that the field should not hardcode a length limit (truncation), even if there is a backend limit - that instead the user's input should be accepted by the input, and a regular validation error thrown.

Comment (edwardhorsford) [reply src/components/password-input/index.md:60]: Context: the guidance is to not limit password lengths, but many legacy systems have them as a requirement. Teams working in such legacy environments may naturally set a max-length on the input thinking they are being helpful, and unaware of the usability issues they are introducing.

Comment (querkmachine) [reply src/components/password-input/index.md]: I've added a paragraph to that effect

**Comment by edwardhorsford** [comment]
@querkmachine that all sounds great. My specific concern was secret questions, but you're right there's other similar things in that category. Perhaps a more generic sentence to state that other password-like things (such as x, y, and z) should not use this pattern, etc.

Super glad to hear that 'show password' is being built in! Great job!

File: src/components/password-input/index.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+Doing this can also prevent ['spell-jacking'](https://www.otto-js.com/news/article/chrome-and-edge-enhanced-spellcheck-features-expose-pii-even-your-passwords), where security researchers have found some spell checking tools gathering personal identifiable information, even user’s passwords, from password input fields to send to third party services.
+
+You can tell browsers not to autocapitalise values by setting the `autocapitalize` attribute to `off`.
+
+## Research on this component
+
+Some browsers, password managers and similar tools will add their own show and hide functionality within the password input field. This might confuse users shown alongside this component’s show and hide button.
+
+We investigated how this component interacts with the similar functionality and worked to prevent this duplication where possible.
```
Comment (calvin-lau-sig7) [inline src/components/password-input/index.md:155]: Draft content only – assumptions is that we'll use this space to summarise conclusions from #3552

**Comment by calvin-lau-sig7** [comment]
@selfthinker added some [new comments in Google Doc](https://docs.google.com/document/d/1J1BphCxT5Rw3xXGbevTmfWSN216Xb0kXTNJUBQAdeiw/edit?usp=sharing) on 19/2, will look to make edits there and add to this PR.

**Review by 36degrees — APPROVED** [review]

**Pull request merged**